File: 53/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum = nums[0]
        tempsum = nums[0]
        for i in range(1, len(nums)):
            tempsum += nums[i]
            tempsum = max(tempsum, nums[i])
            sum = max(sum, tempsum)
        return sum
                
            
        
        

File: 53/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int len = nums.size();
        int ans = nums[0];
        int temp_max = nums[0];
        for(int i=1; i<len; ++i){
            temp_max+=nums[i];
            temp_max = max(temp_max, nums[i]);
            ans = max(temp_max, ans);
        }
        return ans;
    }
};
